py-polars/tests/unit/operations/test_cast.py

```diff
@@ -579,25 +579,29 @@ def test_strict_cast_string_and_binary(
 
 
 @pytest.mark.parametrize(
-    "dtype_out",
+    ("dtype_in", "dtype_out"),
```

Comment:
Could you just make a separate test `test_cast_enum_name_retention`? That should make things a bit cleaner.